src/Microsoft.ML.Core/Data/Repository.cs

```diff
@@ -10,11 +10,6 @@
 
 namespace Microsoft.ML.Model
 {
-    /// <summary>
-    /// Signature for a repository based model loader. This is the dual of ICanSaveModel.
-    /// </summary>
-    public delegate void SignatureLoadModel(ModelLoadContext ctx);
-
     /// <summary>
     /// For saving a model into a repository.
```

Comment:
I'd suggest to mention the two major design patterns introduced in this PR.
(1) A class inherited `ICanSaveModel` from a base class should overwrite the function invoked by `Save(ModelSaveContext ctx)` in that base class.
(2) A class declared as a `ICanSaveModel` should do explicit implementation of `Save(ModelSaveContext ctx)`.

Reply:
I agree, I will do this.

---
In reply to: [254963056](https://github.com/dotnet/machinelearning/pull/2431#discussion_r254963056) [](ancestors = 254963056)

Reply:
Only can add the second part however, because the first part is not general. There can be a base class that knows how to save, and thus simply implements Save directly with no additional abstract function invocation. 

---
In reply to: [255227041](https://github.com/dotnet/machinelearning/pull/2431#discussion_r255227041) [](ancestors = 255227041,254963056)

Reply:
I'd like to have every use documented. It can provide future devs a clear view of your clever idea.

---
In reply to: [255228894](https://github.com/dotnet/machinelearning/pull/2431#discussion_r255228894) [](ancestors = 255228894,255227041,254963056)

Reply:
Sounds good, I added a conditional case at the end of the sentence so that it actually matches all cases.

---
In reply to: [255252654](https://github.com/dotnet/machinelearning/pull/2431#discussion_r255252654) [](ancestors = 255252654,255228894,255227041,254963056)